c_glib/arrow-glib/compute.cpp

```diff
@@ -902,6 +902,43 @@ garrow_double_array_sum(GArrowDoubleArray *array,
                                                      0);
 }
 
+/**
+ * garrow_array_take:
+ * @array: A #GArrowArray.
+ * @indices: The indices of values to take.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: (nullable) (transfer full): The #GArrowArray taken from
+ *   an array of values at indices in input array or %NULL on error.
+ *
+ * Since: 0.14.0
+ */
+GArrowArray *
+garrow_array_take(GArrowArray *array,
+                  GArrowArray *indices,
```

Comment:
Could you add `GArrowTakeOptions *options` to avoid backward incompatibility in the future?

Reply:
I've added it.